NIPostsReload/Presentation/Scenes/NIPostFeedModule/Presenter/NIPostFeedPresenter.swift

```diff
@@ -81,6 +102,48 @@ final class DefaultNIPostFeedPresenter: NIPostFeedPresenter {
         
         view.updateCollectionViewLayout()
     }
+    
+    func searchPosts(query: String?) {
+        searchWorkItem?.cancel()
+        
+        guard let query, !query.isEmpty else {
+            updatePosts()
+            return
+        }
+        
+        guard query.count >= 2 else {
+            return
+        }
+        
+        searchWorkItem = DispatchWorkItem { [unowned self] in
```

Comment:
Using 'unowned self' here might be unsafe. What if the user changes the screen during the search deadline? This could lead to potential issues